jvb/src/main/kotlin/org/jitsi/videobridge/relay/Relay.kt

```diff
@@ -0,0 +1,549 @@
+                // TODO this@Relay.dtlsAppPacketReceived(buf, off, len)
+            }
+        }
+        dtlsTransport.outgoingDataHandler = object : DtlsTransport.OutgoingDataHandler {
+            override fun sendData(buf: ByteArray, off: Int, len: Int) {
+                iceTransport.send(buf, off, len)
+            }
+        }
+        dtlsTransport.eventHandler = object : DtlsTransport.EventHandler {
+            override fun handshakeComplete(
+                chosenSrtpProtectionProfile: Int,
+                tlsRole: TlsRole,
+                keyingMaterial: ByteArray
+            ) {
+                logger.info("DTLS handshake complete")
+                transceiver.setSrtpInformation(chosenSrtpProtectionProfile, tlsRole, keyingMaterial)
+                scheduleRelayMessageTransportTimeout()
+            }
+        }
+    }
+
+    /**
+     * Sets the remote transport information (ICE candidates, DTLS fingerprints).
+     *
+     * @param transportInfo the XML extension which contains the remote
+     * transport information.
+     */
+    fun setTransportInfo(transportInfo: IceUdpTransportPacketExtension) {
+        val remoteFingerprints = mutableMapOf<String, String>()
+        val fingerprintExtensions = transportInfo.getChildExtensionsOfType(DtlsFingerprintPacketExtension::class.java)
+        fingerprintExtensions.forEach { fingerprintExtension ->
+            if (fingerprintExtension.hash != null && fingerprintExtension.fingerprint != null) {
+                remoteFingerprints[fingerprintExtension.hash] = fingerprintExtension.fingerprint
+            } else {
+                logger.info("Ignoring empty DtlsFingerprint extension: ${transportInfo.toXML()}")
+            }
+        }
+        dtlsTransport.setRemoteFingerprints(remoteFingerprints)
+        if (fingerprintExtensions.isNotEmpty()) {
+            val setup = fingerprintExtensions.first().setup
+            dtlsTransport.setSetupAttribute(setup)
+        }
+        iceTransport.startConnectivityEstablishment(transportInfo)
+
+        val websocketExtension = transportInfo.getFirstChildOfType(WebSocketPacketExtension::class.java)
+        if (websocketExtension?.url != null) {
+            _messageTransport.connectTo(websocketExtension.url)
+        }
+    }
+
+    /** Whether [describeTransport] has been called for this relay.
+     */
+    var transportDescribed: Boolean = false
+        private set
+
+    fun describeTransport(): IceUdpTransportPacketExtension {
+        val iceUdpTransportPacketExtension = IceUdpTransportPacketExtension()
+        iceTransport.describe(iceUdpTransportPacketExtension)
+        dtlsTransport.describe(iceUdpTransportPacketExtension)
+        val wsPacketExtension = WebSocketPacketExtension()
+
+        /* TODO: this should be dependent on videobridge.websockets.enabled, if we support that being
+         *  disabled for relay.
+         */
+        if (_messageTransport.isActive) {
```

Comment:
I think we should stop adding the "active" flag in signaling, as it's not actually used and the semantics are getting confusing. I think `_messageTransport.isActive` means "I have been told to connect to remote relay". We describe the transport only when responding to the "create" message (I just pushed that change), anyway, so we can simply always describe the WS with the URL.

Reply:
I think the case this is meant to handle is if the initial create message actually included the peer's transport (i.e. you're doing an offer/answer-style negotiation rather than the dual-create style), then there's no need for us to send the URL.  But maybe that's unnecessary since you're not planning on using that mode in jicofo.